rgw/admin/bucket.go

```diff
@@ -96,6 +96,66 @@ func (api *API) ListBuckets(ctx context.Context) ([]string, error) {
 	return s, nil
 }
 
+// BucketListingSpec describes a request
+type BucketListingSpec struct {
+	UID          string `url:"uid"`
+	GenerateStat *bool  `url:"stats"`
+}
+
+// ListUsersBuckets will return the list of all users buckets without stat
+//  PREVIEW
```

Comment:
Sorry if the docs don't currently make this clear but new functions added as "PREVIEW" must also be "protected" by a build tag "ceph_preview". Unfortunately, this means you can't currently add a new API function to an existing file with stable functions in it.

Also tagging @leseb for attention for these rgw api changes.

Reply:
@phlogistonjohn thank you for the feedback. I've added commit, moving new API methods to separate files with `+build ceph_preview`